### PR TITLE
Fix #4473 Repository validator doesn't validate ref attribute value types

### LINKED ISSUE #4473: Repository validator does not validate ref attribute value types
Reproduce:
- As a developer add or update an entity with 1) a ref attribute where the value is a entity of the wrong type or 2) a mref attribute where the value is a iterable containing null values

Expected:
- Validation error is thrown by the validation decorator

Actual:
- Depends on the entity implementation, silently returns null for some implementations

### PR DESCRIPTION
**Pull request by dennishendriksen**

### CODE REVIEW

File: molgenis-data-validation/src/main/java/org/molgenis/data/validation/EntityAttributesValidator.java [hunk 1]
```diff
@@ -97,6 +114,58 @@
...
+		if (refEntities == null)
+		{
+			return createConstraintViolation(entity, attr, meta, "Not a valid entity, expected an entity list.");
+		}
+		for (Entity refEntity : refEntities)
+		{
+			if (refEntity == null)
+			{
+				return createConstraintViolation(entity, attr, meta, "Not a valid entity, null is not allowed");
+			}
```
Comment (tommydeboer) [inline molgenis-data-validation/src/main/java/org/molgenis/data/validation/EntityAttributesValidator.java]: https://github.com/molgenis/molgenis/issues/4336#issuecomment-178492639 :
> The behavior for requesting a single entity by id is returning null. The behavior of requesting multiple entities by id is thus consistent and I propose to a) keep it that way b) revert the ElasticSearchRepository.findAll() fix c) mention this behavior in the DataService and Repository docs.

Didn't we decide that null *is* allowed/required when requesting multiple entities?

**Comment by tommydeboer** [comment]
test this please

**Comment by tommydeboer** [comment]
test this please

**Comment by Mark-de-Haan** [comment]
test this please

**Pull request merged**